parquet-hadoop/src/main/java/parquet/hadoop/MemoryManager.java

```diff
@@ -47,6 +47,7 @@
   private final long minMemoryAllocation;
   private final Map<InternalParquetRecordWriter, Long> writerList = new
       HashMap<InternalParquetRecordWriter, Long>();
+  private final Map<String, Runnable> callBacks = new HashMap<String, Runnable>();
```

Comment:
Yes, it is good for callback to be able to get values of MemoryManager. Using `Callable<T>` is ok. Besides that, I was thinking public a method like `getCurrentScale` in MemoryManger, so that cackback could get wanted value in its implementation through MemoryManger instance. 

How does this sound? If this way is also ok, shall we work it in a follow on PR?